### PR TITLE
fix: active wrong when submenus has key ''

### PR DESCRIPTION
**Pull request by coderz-w**
close https://github.com/react-component/menu/issues/719
close https://github.com/ant-design/ant-design/issues/49106

### CODE REVIEW

File: src/hooks/useKeyRecords.ts [hunk 1]
```diff
@@ -73,15 +73,18 @@ export default function useKeyRecords() {
...
+      }
+      return pathKeys.some(pathKey => {
         const pathKeyList = getKeyPath(pathKey, true);
 
         return pathKeyList.includes(eventKey);
-      }),
+      });
+    },
     [getKeyPath],
   );
```
Comment (afc163) [inline src/hooks/useKeyRecords.ts:87]:
```suggestion
  const isSubPathKey = useCallback(
    (pathKeys: string[], eventKey: string) =>
      pathKeys.filter(item => item !== undefined).some(pathKey => {
        const pathKeyList = getKeyPath(pathKey, true);
        return pathKeyList.includes(eventKey);
      }),
    [getKeyPath],
  );
```

Comment (afc163) [reply src/hooks/useKeyRecords.ts:87]: 这样是不更简单点？

**Comment by afc163** [comment]
需要补充用例。

**Comment by coderz-w** [comment]
> 需要补充用例。

好，佬之前那个看一下https://github.com/react-component/upload/pull/559

File: tests/SubMenu.spec.tsx [hunk 2]
```diff
@@ -149,6 +149,30 @@ describe('SubMenu', () => {
     ).textContent;
     expect(childText).toEqual('submenu');
   });
+  it("The submenu item with key '' must not persistently remain active.", () => {
```
Comment (afc163) [inline tests/SubMenu.spec.tsx:152]:
```suggestion

  // https://github.com/ant-design/ant-design/issues/49106
  it('The submenu item with key '' must not persistently remain active', () => {
```

File: tests/SubMenu.spec.tsx [hunk 3]
```diff
@@ -149,6 +149,30 @@ describe('SubMenu', () => {
...
+              {
+                label: 'Navigation One1',
+                key: 'mail1',
+              },
+            ],
+          },
+        ]}
+      />,
+    );
+    expect(container.querySelector('.rc-menu-submenu-active')).toBeFalsy()
```
Comment (afc163) [inline tests/SubMenu.spec.tsx:174]:
```suggestion
    expect(container.querySelector('.rc-menu-submenu-active')).toBeFalsy();
```

File: tests/SubMenu.spec.tsx [hunk 4]
```diff
@@ -150,6 +150,32 @@
     expect(childText).toEqual('submenu');
   });
 
+  // https://github.com/ant-design/ant-design/issues/49106
+  it('The submenu item with key '' must not persistently remain active', () => {
```
[the remark this answers is not in the document]
Comment (afc163) [reply tests/SubMenu.spec.tsx:154]: 语法有错误。

File: tests/SubMenu.spec.tsx [hunk 5]
```diff
@@ -149,6 +149,30 @@ describe('SubMenu', () => {
     ).textContent;
     expect(childText).toEqual('submenu');
   });
+  it(`The submenu item with key '' must not persistently remain active`, () => {
```
Comment (afc163) [inline tests/SubMenu.spec.tsx:152]:
```suggestion

  it(`The submenu item with key '' must not persistently remain active`, () => {
```

**Pull request merged**